### PR TITLE
Switch to panic='abort' for safety across FFI boundaries

### PR DESCRIPTION
**Pull request by ValarDragon**
Thanks to @daira for pointing out the security impact, and @jon-chuang for raising the issue / pointing out perf improvements.

This still needs to be done in downstream repositories, as this change only applies to the top level crate.

closes: #183
---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests - n/a?
- [ ] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by Pratyush** [comment]
Doesn’t this only affect our libraries, and not any downstream binaries?

**Comment by ValarDragon** [comment]
Yes. my understanding of the [profile docs](https://doc.rust-lang.org/cargo/reference/profiles.html) is that this will only take precedence for FFI that is obtained from compiling `algebra` directly. (E.g. golang code using FFI for this curve arithmetic, and implementing BLS with it)

I don't think this is a big concern for algebra, but is much more likely to be one for groth16, and it doesn't really hurt to do it in all libraries.

File: Cargo.toml [hunk 1]
```diff
@@ -30,6 +31,7 @@ debug-assertions = false
 
 [profile.dev]
 opt-level = 0
+panic = 'abort'
```
Comment (Pratyush) [inline Cargo.toml:34]: Is there a benefit to turning it to `abort` here?

Comment (ValarDragon) [reply Cargo.toml:34]: Not to my knowledge, but it feels like a safer default imo

**Review by Pratyush — APPROVED** [review]

**Pull request merged**